forte/mrdsrg-spin-integrated/dsrg_mrpt2_grad/dsrg_mrpt2_deriv_tensors.cc

```diff
@@ -0,0 +1,515 @@
+
+using namespace ambit;
+using namespace psi;
+
+namespace forte {
+
+/**
+ * Initializing CI integrals for the z-vector equation.
+ */
+void DSRG_MRPT2::set_ci_ints() {
+    cc = coupling_coefficients_;
+    ci = ci_vectors_[0];
+
+    cc1a_n = ambit::Tensor::build(ambit::CoreTensor, "cc1a_n", {ndets, na, na});
+    cc1a_r = ambit::Tensor::build(ambit::CoreTensor, "cc1a_r", {ndets, na, na});
+    cc1b_n = ambit::Tensor::build(ambit::CoreTensor, "cc1b_n", {ndets, na, na});
+    cc1b_r = ambit::Tensor::build(ambit::CoreTensor, "cc1b_r", {ndets, na, na});
+
+    cc2aa_n = ambit::Tensor::build(ambit::CoreTensor, "cc2aa_n", {ndets, na, na, na, na});
+    cc2aa_r = ambit::Tensor::build(ambit::CoreTensor, "cc2aa_r", {ndets, na, na, na, na});
+    cc2bb_n = ambit::Tensor::build(ambit::CoreTensor, "cc2bb_n", {ndets, na, na, na, na});
+    cc2bb_r = ambit::Tensor::build(ambit::CoreTensor, "cc2bb_r", {ndets, na, na, na, na});
+    cc2ab_n = ambit::Tensor::build(ambit::CoreTensor, "cc2ab_n", {ndets, na, na, na, na});
+    cc2ab_r = ambit::Tensor::build(ambit::CoreTensor, "cc2ab_r", {ndets, na, na, na, na});
+
+    cc3aaa_n = ambit::Tensor::build(ambit::CoreTensor, "cc3aaa_n", {ndets, na, na, na, na, na, na});
+    cc3aaa_r = ambit::Tensor::build(ambit::CoreTensor, "cc3aaa_r", {ndets, na, na, na, na, na, na});
+    cc3bbb_n = ambit::Tensor::build(ambit::CoreTensor, "cc3bbb_n", {ndets, na, na, na, na, na, na});
+    cc3bbb_r = ambit::Tensor::build(ambit::CoreTensor, "cc3bbb_r", {ndets, na, na, na, na, na, na});
+    cc3aab_n = ambit::Tensor::build(ambit::CoreTensor, "cc3aab_n", {ndets, na, na, na, na, na, na});
+    cc3aab_r = ambit::Tensor::build(ambit::CoreTensor, "cc3aab_r", {ndets, na, na, na, na, na, na});
+    cc3abb_n = ambit::Tensor::build(ambit::CoreTensor, "cc3abb_n", {ndets, na, na, na, na, na, na});
+    cc3abb_r = ambit::Tensor::build(ambit::CoreTensor, "cc3abb_r", {ndets, na, na, na, na, na, na});
+
+    cc1a_n("Kzw") = cc.cc1a()("KJzw") * ci("J");
+    cc1a_r("Kzw") = cc.cc1a()("JKzw") * ci("J");
+    cc1b_n("KZW") = cc.cc1b()("KJZW") * ci("J");
+    cc1b_r("KZW") = cc.cc1b()("JKZW") * ci("J");
+
+    cc2aa_n("Kuvxy") = cc.cc2aa()("KJuvxy") * ci("J");
+    cc2aa_r("Kuvxy") = cc.cc2aa()("JKuvxy") * ci("J");
+    cc2bb_n("KUVXY") = cc.cc2bb()("KJUVXY") * ci("J");
+    cc2bb_r("KUVXY") = cc.cc2bb()("JKUVXY") * ci("J");
+    cc2ab_n("KuVxY") = cc.cc2ab()("KJuVxY") * ci("J");
+    cc2ab_r("KuVxY") = cc.cc2ab()("JKuVxY") * ci("J");
+
+    cc3aaa_n("Kuvwxyz") = cc.cc3aaa()("KJuvwxyz") * ci("J");
+    cc3aaa_r("Kuvwxyz") = cc.cc3aaa()("JKuvwxyz") * ci("J");
+    cc3bbb_n("KUVWXYZ") = cc.cc3bbb()("KJUVWXYZ") * ci("J");
+    cc3bbb_r("KUVWXYZ") = cc.cc3bbb()("JKUVWXYZ") * ci("J");
+    cc3aab_n("KuvWxyZ") = cc.cc3aab()("KJuvWxyZ") * ci("J");
+    cc3aab_r("KuvWxyZ") = cc.cc3aab()("JKuvWxyZ") * ci("J");
+    cc3abb_n("KuVWxYZ") = cc.cc3abb()("KJuVWxYZ") * ci("J");
+    cc3abb_r("KuVWxYZ") = cc.cc3abb()("JKuVWxYZ") * ci("J");
+
+    dlamb_aa =
```

Comment:
If ever you are referring to a specific equation from your paper (and I think you are here?), please give the exact number for reference.

Reply:
Good point. I leave comments like "derivatives of Lambda3_ w.r.t. C_K alpha-alpha-alpha", which clearly identifies what those equations are for. I think information like that is sufficient since these equation terms won't get changed in the future maintenance, as the definition of density or cumulant tensors is not changed.

Reply:
And for this section, that's fine (assuming this is the reference cumulant?), but what about all the other equations?